static/scenes/railroad/js/present.js

```diff
@@ -61,9 +62,16 @@ class Present {
         }
 
         parent.add(this.model);
+
+        this.shootPromise = undefined;;
+        this.shootResolveFunction = undefined;
     }
 
-    shoot(targetPosition) {
+    async shoot(targetPosition) {
+        if (this.shootPromise !== undefined) {
+            return this.shootPromise;
+        }
```

Comment:
Added this extra check to avoid shooting the present more than once. I am thinking this aligns with the way presents behave but if we want to shoot a present more than once let me know and I can change it.